testfile.alc

```diff
@@ -38,11 +40,13 @@ let array-type =
 let (val) = (prim-sub 5 1)
 prim-sub val 3
 
-#let number-array = array-type (box prim-type prim-number)
+let number-array = (array-type (box prim-type prim-number))
 
 let sub = 
 	lambda (x : prim-number, y : prim-number)
 		let (val) = (prim-sub x y)
```

Comment:
x is no longer in the env on this line which is wrong :ded:

@ErikMcClure can you try debugging this